bin/onchange-nfs.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+if [[ `curl -s ${CONSUL}:8500/v1/health/state/passing | grep nfs`  ]]
+then
+  echo "nfs is healthy, mounting uploads directory...."
+  mount -t nfs -v -o nolock,vers=3 nfs:/exports /var/www/html/content/uploads
+else
```

Comment:
Wouldn't it break in much the same way if the NFS container suddenly goes away without unmounting? But I guess in the case of unmounting it like this, a user could suddenly start writing uploads into the /var/www/content/uploads directory locally which will cause all manner of hell once the NFS container comes back up....

I am looking into ways that we can disable file uploads if the container were to go away. There's nothing in WP itself that will let you disable uploads, save some code we could put in a theme or plugin to remove the capability from a certain user class (all of them?)

I'm toying with the idea of creating another consul template to create an included configfile (think memcached-config.php) that would redefine the uploads directory to a "fallback_uploads" directory local on the container. If NFS is up, this config file would be empty, if it is not then it simply contains "define('uploads', '/var/www/html/content/fallback-uploads/');. Then maybe in our  onchange-nfs.sh script we rsync (would need to add that to the image) everything from fallback-uploads to uploads and then delete stuff from fallback-uploads.

On first glace, this sounds like it might even play nicely with the read caching you mentioned above, but after writing it out and thinking on it more there are some more issues that play, this would redefine how wordpress generates the src on img tags, and would probalby wreak all kinds of caching havoc.

I'm not as convinced by this idea anymore, but I'll leave all that there as a stream-of-consciousness note and keep thinking on this problem.